pages/product/[code].js

```diff
@@ -0,0 +1,178 @@
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import Image from "next/image";
+import styled from "styled-components";
+import { atomWithStorage, createJSONStorage } from "jotai/utils";
+import { useAtom } from "jotai";
+import allergens from "../../allergens.json";
+import additives from "../../additives.json";
+
+const StyledImage = styled(Image)`
+  width: 25%;
+  height: 25%;
+  object-fit: cover;
+`;
+//getting additives from Localstorage with atom from jotai
+const initialAdditives = atomWithStorage("additives", [], {
+  ...createJSONStorage(() => localStorage),
+  delayInit: true,
+});
+//getting  allergens from Localstorage with atom from jotai
+const initialAllergens = atomWithStorage("allergens", [], {
+  ...createJSONStorage(() => localStorage),
+  delayInit: true,
+});
+export default function DetailPage() {
+  const [additivesFromStorage] = useAtom(initialAdditives);
+  const [allergensFromStorage] = useAtom(initialAllergens);
+
+  const router = useRouter();
+  const { code } = router.query;
+  const { data } = useSWR(
+    `https://de.openfoodfacts.org/api/v0/product/${code}.json`
+  );
+
+  function Additives({ additive }) {
+    return <div>{additive}</div>;
+  }
+
+  function BackToScanner() {
+    return (
+      <button
+        onClick={() => {
+          router.push("/barcodeScanner");
+        }}
+      >
+        <span>◀︎</span>
+      </button>
+    );
+  }
+  if (!data) {
+    return (
+      <>
+        <BackToScanner />
+        <p>loading</p>
+      </>
+    );
+  }
+  if (!data.product) {
+    return (
+      <>
+        <BackToScanner />
+        <p>Product not found</p>
+      </>
+    );
+  }
+  //converting the fetched allergensString to an array of
+  //strings and than map to an array of objects
+
+  const allergensArray = data.product.allergens.split(",").map((item) => ({
+    id: item,
+  }));
+
+  const additivesArray = data.product.additives_original_tags.map((item) => ({
+    id: item,
+  }));
+
+  function Allergens() {
+    return (
+      <div>
+        {allergensArray.map((item) => (
+          <div key={item.id}>
+            {allergens.tags.find((aller) => aller.id === item.id)?.name}
+          </div>
+        ))}
+      </div>
+    );
+  }
+
+  const filteredAdditives = additivesFromStorage.filter((additive) => {
+    return additivesArray.some((addi) => addi.id === additive.id);
+  });
+
+  const filteredAllergens = allergensFromStorage.filter((allergen) => {
+    return allergensArray.some((allerg) => allerg.id === allergen.id);
+  });
+
+  return (
+    <>
+      <button
+        onClick={() => {
+          router.push("/barcodeScanner");
+        }}
+      >
+        <span>◀︎</span>
+      </button>
```

Comment:
Why not use your function ```BackToScanner``` above? 🙂

Reply:
i don't know why this happend .i thought i implemented it overall ;) i think i will update it with next pr or maybe now